### ISSUE TITLE
[gtk3] Can not type passprhasse, wrong terminal focus, can not login to any server

### ISSUE THREAD

**Issue by hanspr** [asbru-cm/asbru-cm#204]
My terminal configuration

![imagen](https://user-images.githubusercontent.com/1572396/66609192-05f72380-eb7e-11e9-8520-5fb374f33a08.png)

1.- Double clic to open a session
2.- Window will show up with no focus

![imagen](https://user-images.githubusercontent.com/1572396/66609298-48b8fb80-eb7e-11e9-8d59-09f4c77e2555.png)

If I type the passphrasse it will show instead of being hidden

![imagen](https://user-images.githubusercontent.com/1572396/66609322-58d0db00-eb7e-11e9-929d-47401ec13141.png)

After typing the the connection is lost, no passphrasse ever transmitted

You can only connect if your key has no passphrase

**Desktop (please complete the following information):**
 - Linux Mint
 - 18.3

**Comment by hanspr** [comment]
More tests

Fresh install, no configurations changed, only added one connection by hand and on connection, it opens a new tab, no focus, no passphrase entered

![imagen](https://user-images.githubusercontent.com/1572396/66610456-368c8c80-eb81-11e9-9af3-52b59e277604.png)

If you keep watting it ends with a disconnect error

![imagen](https://user-images.githubusercontent.com/1572396/66610487-499f5c80-eb81-11e9-8c02-fabdd12408bd.png)

**Comment by hanspr** [comment]
I manged to make it work with this regex

([pP]ass|[pP]ass[wW]or[dt](\s+for\s+|\w+@\w+)*|[cC]ontrase.a|Enter passphrase for key ).*:.*$

But now it asks me for my passphrase with a popup window

![imagen](https://user-images.githubusercontent.com/1572396/66611988-d4825600-eb85-11e9-8d90-d80b95b88d0a.png)

**Comment by hanspr** [comment]
Managed to fix this for me, I don't know why are you forcing this window on publickeys?

Removed ask for password on public keys

![imagen](https://user-images.githubusercontent.com/1572396/66612686-3c39a080-eb88-11e9-9209-358e07771561.png)

Or I'm missing a new configuration option?

**Comment by hanspr** [comment]
The only remaining problem is focuse again

**Comment by hanspr** [comment]
Managed to fix focus once and for all too.

Added this line.

![imagen](https://user-images.githubusercontent.com/1572396/66614337-4d85ab80-eb8e-11e9-9748-f54bc01a9485.png)

`$PACMain::RUNNING{ $$self{'_UUID_TMP'} }{terminal}{_GUI}{_VTE} -> grab_focus;`

**Comment by gfrenoy** [comment]
Hans, many things in this issue ; try to keep 1 topic per issue, it will be easier to follow-up.

So, let's try:

1) The way to capture a password is indeed via a popup
2) The passphrase is only supposed to be asked if you specified a SSH key (hence the ```$AUTH eq 'publickey'``` test)
3) What is wrong with the current password regular expression ?
4) As we have 2, it is not supposed to grab the focus at this stage, it is done later on.

All in all, it seems to me that one the password regexp is not working for you ; we should focus on that issue only.  The other workarounds may work for you but are likely not necessary if we fix your root issue.

**Comment by hanspr** [comment]
Hi Gaëtan,

In the previous version, and PAC, you could. Select a key, type a user and a password. Like this.

![imagen](https://user-images.githubusercontent.com/1572396/66700904-a141e300-ecbb-11e9-9a1e-de2a6b358941.png)

And then the full login would be authomated.

At some point you added that and forcing that a passphrase must be entered on the box, even if the user decided to typed in the password box.

So, I understand that you consider this unsafe and decided that is obligatory to type a password?, is that the reason behind the modification. IMHO, I think that the popup should appear only if a user leaves the password blank, to indicate that he/she wishes to imput a password, and if it is typed, then it means you want to save it in Asbru and automate the full login.

With respect to the regular expression, this is the original one:

([pP]ass|[pP]ass[wW]or[dt](\\s+for\\s+|\\w+@\\w+)*|[cC]ontrase.a|Enter passphrase for key )\\s*:\\s*$

And the expect line to match is this

Enter passphrase for key '/home/hans/Documentos/Seguro/llaves/sanfandila/eflow.openssh':

The regular experssion expects key[SPACE]) and it receives key[quote]

So the only way to make it work is like this

([pP]ass|[pP]ass[wW]or[dt](\s+for\s+|\w+@\w+)*|[cC]ontrase.a|Enter passphrase for key).*:.*$

**Comment by gfrenoy** [comment]
> In the previous version (...) select a key, type a user and a password (...)

Ah ! That is something else :)  I indeed only tested *without* a password and in that case, it's the expected behavior.

Let me check when a passphrase is given.

> With respect to the regular expression, this is the original one: (...)

Oh I understand the problem; the original one *should* be:

```([pP]ass|[pP]ass[wW]or[dt](\s+for\s+|\w+@\w+)*|[cC]ontrase.a|Enter passphrase for key '.+')\s*:\s*$```

But in the code, the quote is wrongly escaped :-/  See [here](https://github.com/asbru-cm/asbru-cm/blob/master/lib/PACUtils.pm#L1816).

Let me see if I can do something about this.

**Comment by gfrenoy** [comment]
> In the previous version (...) select a key, type a user and a password (...)

What I did [here](https://github.com/asbru-cm/asbru-cm/commit/39986625a71b571d803f6717fb22016f7029e415) was not working when you did specify a passphrase.

I reviewed the condition to better manage this !  Hopefully it is correct now.  The idea is:

- If <<ASK_PASS>> is given, we do ask for the password
- for the "publickey" method, If no password is given (or if password is empty) ; it should be accepted and not prompted so if we receive a prompt, we ask for the password

Let me know if now works for you.

> With respect to the regular expression, this is the original one: (...)

A unwanted regression occurred [here](https://github.com/asbru-cm/asbru-cm/commit/d3b14a5089caf61a180252937f8cd594d8b606d1).

It should be fixed now for new configuration files / new connections.

You'll have to change it in your existing configuration manually though...
